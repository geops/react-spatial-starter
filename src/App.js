import React from 'react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';
import Root from './components/Root';

import store from './model/store';

import './App.scss';

const propTypes = {
  history: PropTypes.oneOfType([ReactRouterPropTypes.history]),
  initialState: PropTypes.object,
};

const defaultProps = {
  history: null,
  initialState: {},
};

const App = ({ history, initialState }) => (
  <Provider store={store}>
    <div className="app">
      <Root history={history} initialState={initialState} />
    </div>
  </Provider>
);

App.propTypes = propTypes;
App.defaultProps = defaultProps;

export default App;
