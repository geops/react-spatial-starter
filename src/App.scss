body {
  font-family: 'SBBWeb Roman', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 15px;
  margin: 0;
}

.tm-app {
  height: 100%;
  width: 100%;
  position: absolute;
  overflow: hidden;
}

.tm-map {
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
}

h1 {
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
}
