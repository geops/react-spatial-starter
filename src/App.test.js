import 'jest-canvas-mock';
import React from 'react';
import ReactDOM from 'react-dom';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from './App';

configure({ adapter: new Adapter() });

describe('App', () => {
  test('App should match snapshot.', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('App should render div.', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
