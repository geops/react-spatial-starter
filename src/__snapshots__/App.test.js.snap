// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should match snapshot. 1`] = `
<div
  className="app"
>
  <div
    className="tm-app-component"
  >
    <div
      className="tm-map"
      role="menu"
      tabIndex={0}
    />
  </div>
</div>
`;
